contracts/DomainRegistry.sol

```diff
@@ -59,42 +62,54 @@ contract DomainRegistry is Ownable {
     }
 
     modifier checkDomainReleasing(string memory domainName) {
-        string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
-        require(domains[domainName] != address(0), "Domain is not registered yet");
-        require(domains[domainName] == msg.sender, "Domain should be unregistered by the domain owner");
+        require(_domains[domainName] != address(0), "Domain is not registered yet");
+        require(_domains[domainName] == msg.sender, "Domain should be unregistered by the domain owner");
         _;
     }
 
     function getDomain(string memory domainName) public view returns (address) {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
 
-        return domains[rootDomain];
+        return _domains[rootDomain];
     }
 
-    function registerDomain(string memory domainName) external payable
-        checkDomainLength(domainName)
+    function validateDomainRegistration(string memory domainName) internal
         checkDomainParent(domainName)
         checkDomainAvailability(domainName)
         checkSufficientEtn(domainName)
+    {
+    }
+
+    function withdraw() external {
+        uint share = _shares[msg.sender];
+        _shares[msg.sender] = 0;
+        payable(msg.sender).transfer(share);
```

Comment:
By this moment the `_shares` mapping looks excessive. You have only one contract owner. Yes, the `OwnableUpgradable` functionality allow you to change the owner address but it is not enough reason to keep ether on each of the historical owner addresses. It complicates contract management more than benefits it brings. Apart from that `_shares` name does not say as much as, at least, `contractOwnersBalances`
Much better would be not to have any state var at all. Just withdraw all the contract balance. It is simple, enough for the task and works even when contract owner changes (just withdraw before transfering ownership)
      -2 points